subprojects/core-api/src/main/java/org/gradle/api/tasks/util/internal/PatternSpecFactory.java

```diff
@@ -85,26 +87,25 @@
         String[] defaultExcludes = DirectoryScanner.getDefaultExcludes();
         if (defaultExcludeSpecCache.isEmpty()) {
```

Comment:
Can we simplify this logic now? So if the cache is empty we populate it, and if it isn't, we verify that the default excludes didn't change. We don't need to update the spec cache the default excludes didn't change and the cache is non-empty, right?